@here/harp-mapview/lib/SkyBackground.ts

```diff
@@ -70,9 +70,13 @@ export class SkyBackground {
      * Updates the sky texture with new parameters.
      *
      * @param params New sky configuration parameters.
+     * @param projectionType Which projection is used, this may also change (in which case the
+     * textures should be recreated).
      */
-    updateTexture(params: GradientSky | CubemapSky) {
-        const isSameSkyType = this.m_sky.type === params.type;
+    updateTexture(params: GradientSky | CubemapSky, projectionType: ProjectionType) {
+        const isSameSkyType =
+            this.m_sky.type === params.type && this.m_projectionType === projectionType;
+        this.m_projectionType = projectionType;
```

Comment:
I'd move this line to the end of `updateTexture` (to be next to `this.m_sky`)

Reply:
I could, but then I would need to change line 85, and it would make it slightly more complicated.
Maybe I could instead move this.m_sky to be closer to this (I don't see the need to have it at the bottom).

Reply:
@nzjony One reason I could tell it should be in the bottom is that if for example you provide a wrong projectionType, or any other thing will go wrong, then it will throw an error and exit a function, and changes won't be applied (`this.m_sky` & `this.m_projectionType` will be the same as before) and it should be normal behavior.

Reply:
Done